messaging/src/main/java/org/axonframework/queryhandling/SimpleQueryBus.java

```diff
@@ -369,6 +438,49 @@ public Registration registerDispatchInterceptor(
                             .collect(Collectors.toList());
     }
 
+    private <Q, R> Publisher<MessageHandler<? super QueryMessage<?, ?>>> getStreamingHandlersForMessage(
+            StreamingQueryMessage<Q, R> queryMessage) {
+        ResponseType<?> responseType = queryMessage.getResponseType();
+        return Flux.fromStream(subscriptions.computeIfAbsent(queryMessage.getQueryName(), k -> new CopyOnWriteArrayList<>())
+                                            .stream()
+                                            .filter(subscription -> responseType.matches(subscription.getResponseType()))
+                                            .sorted(new MessageHandlerComparator())
+                                            .map(QuerySubscription::getQueryHandler)
+                                            .map(queryHandler -> (MessageHandler<? super QueryMessage<?, ?>>) queryHandler));
+    }
+
+    /**
+     * Comparator to prioritize handlers based on the response type.
```

Comment:
This comparator is specific for the `streamingQuery`, right? If so, might be nice to add that's the case right now, to quickly rule out any future assumptions.

Reply:
Once https://github.com/AxonFramework/AxonFramework/pull/2148 is merged, I guess we can remove this comparator :)